sklearn/preprocessing/_polynomial.py

```diff
@@ -1003,33 +1045,56 @@ def transform(self, X):
                     x = spl.t[spl.k] + (X[:, i] - spl.t[spl.k]) % (
                         spl.t[n] - spl.t[spl.k]
                     )
-                else:
+                else:  # self.extrapolation in ("continue", "error")
                     x = X[:, i]
 
                 if use_sparse:
+                    x = x.copy()  # copy to avoid inplace operation
+                    # we leave the nan values out of bsplining and make them into 0s
+                    # right away, since BSpline.design_matrix() would raise on the nans
+                    # when X is sparse:
+
+                    x[nan_indicator[:, i]] = np.nanmin(x)
```

Comment:
We could use `np.nanmin(X)` instead of 0, I believe, but then there is another problem: `spl.t` is nans as well and `BSpline.design_matrix()`  will still raise.